README.md

```diff
@@ -62,70 +84,122 @@ highlighted](https://i.imgur.com/XN7xdKF.png)
 *Example: An element which occupies half the viewport shifts by a distance equal
 to half its height.  The impact fraction for this animation frame is 0.75.*
 
-In general, the layout shift (LS) score is equal to the impact fraction.
+### Distance Fraction
+
+The **move distance** of a shifting element is the distance it has moved on
+the horizontal or vertical axis (whichever is greater).
+
+The **distance fraction** of an animation frame is the greatest move distance
+of any shifting element in that frame, divided by the width or height
+(whichever is greater) of the viewport.
+
+![Illustration of shifting elements on a device, with their move distances
+indicated by arrows](https://i.imgur.com/qeks8UK.png)
+
+*Example: The most-shifted element moved a distance of one quarter of the
+viewport.  The distance fraction for this animation frame is 0.25.*
+
+The intent of incorporating the distance fraction into the LS score calculation
+is to avoid overly penalizing cases where large elements shift by small
+distances.
 
-TODO(skobes): describe distance fraction proposal
+### LS Score Calculation
 
-However, if the user has generated certain
-[UI events](https://www.w3.org/TR/uievents/) within the past 500 ms, the LS
-score is 0.  This allows the page to modify its layout in response to the
-event.  The event types that trigger this exception include taps, key presses,
-and mouse clicks, but not mousemove or events that cause scrolling.
+The layout shift (LS) score is equal to the impact fraction multiplied by the
+distance fraction.
 
 The user agent may trade off precision for efficiency in the computation of
 LS scores.  It is intended that the LS score have a correspondence to the
 perceptual severity of the instability, but not that all user agents produce
 exactly the same LS scores for a given page.
 
+### Performance API
+
+Animation frames with non-zero LS scores will notify a registered
+[PerformanceObserver](https://w3c.github.io/performance-timeline/#the-performanceobserver-interface).
+The observer's callback receives one or more `LayoutShift` entries:
+
+```idl
+interface LayoutShift : PerformanceEntry {
+    readonly attribute double value;
+    readonly attribute boolean hadRecentInput;
+    readonly attribute DOMHighResTimeStamp lastInputTime;
+};
+```
+
+The entry's `value` attribute is the LS score.  Its
+[entryType](https://w3c.github.io/performance-timeline/#dom-performanceentry-entrytype)
+attribute is `"layout-shift"`.
+
+The `hadRecentInput` and `lastInputTime` attributes are described in
+[Recent Input Exclusion](#Recent-Input-Exclusion).
```

Comment:
Does this link work?

Reply:
For me it does, do you see something different?